internal/server/server.go

```diff
@@ -0,0 +1,74 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/sarahhtbarton/microsoftleap/internal/helper"
+)
+
+func NewServer(addr string) (*http.Server) {
+	
+	r := NewHTTPHandler()
+
+	return &http.Server{
+		Addr:    addr,
+		Handler: r,
+	}
+}
+
+func NewHTTPHandler() (http.Handler) {
+
+	r := mux.NewRouter()
+
+	r.HandleFunc("/", handler).Methods("GET")
+
+	return r
+}
+
+func ParseMatrixDimensions(vars map[string][]string) (int, int, error) {
+
+	if len(vars["rows"]) == 0 || len(vars["columns"]) == 0 {
```

Comment:
nit: a more idiomatic way of making this check would be to directly compare against the string zero value. `vars["rows"] == ""`

Reply:
@AYM1607 var["rows"] gives you a _slice_ with a string, instead of a string directly.

so if vars := map[string][]string{"rows": {"3"}, "columns": {"5"}}
vars["rows"] = ["3"]
vars["rows"][0] = "3"

does using `len()` make sense or is there still a more idiomatic way of checking?

Reply:
You're right, I missed that, len makes sense in this case.